utils.go

```diff
@@ -0,0 +1,23 @@
+package armbalancer
+
+import (
+	"strings"
+)
+
+func getTransportHostToCompare(reqHost, transportHost string) string {
```

Comment:
I'd prefer to keep this in armbalancer.go and make it a member of `recyclableTransport` so we can reference `t.host` directly without needing to pass it as an arg.

Reply:
I found it will have race condition to read and update t.host by multiple go routines during tests.

Reply:
no worries, found a way that no need to update t.host